lib/matplotlib/tests/test_cbook.py

```diff
@@ -680,14 +680,37 @@ def test_reshape2d_pandas(pd):
     for x, xnew in zip(X.T, Xnew):
         np.testing.assert_array_equal(x, xnew)
 
+
+def test_reshape2d_xarray(xr):
+    # separate to allow the rest of the tests to run if no xarray...
     X = np.arange(30).reshape(10, 3)
-    x = pd.DataFrame(X, columns=["a", "b", "c"])
+    x = xr.DataArray(X, dims=["x", "y"])
     Xnew = cbook._reshape_2D(x, 'x')
     # Need to check each row because _reshape_2D returns a list of arrays:
     for x, xnew in zip(X.T, Xnew):
         np.testing.assert_array_equal(x, xnew)
 
 
+def test_index_of_pandas(pd):
+    # separate to allow the rest of the tests to run if no pandas...
+    X = np.arange(30).reshape(10, 3)
+    x = pd.DataFrame(X, columns=["a", "b", "c"])
+    Idx, Xnew = cbook.index_of(x)
+    np.testing.assert_array_equal(X, Xnew)
+    IdxRef = np.arange(10)
+    np.testing.assert_array_equal(Idx, IdxRef)
+
+
+def test_index_of_xarray(xr):
```

Comment:
Does xarray get us more coverage here? They have a `to_numpy()` method the same as pandas I believe.
https://xarray.pydata.org/en/stable/generated/xarray.DataArray.to_numpy.html

So, it seems like a pretty heavy dependency to add for just this one test...

Reply:
Not so much for coverage as for actually testing using data of specified formats. With the discussion about which formats we support, it makes sense to test them as well. Right now some of these are tested in the plots, but it can possibly make sense to simply test them here as these are the core function used to get data that can be plotted. 

If we claim (which we actually don't, maybe we should?) that we can plot xarray, we should probably test it as well. And other types that we may want to claim to support. Or maybe fork off a specific dependency test that is not executed on all platforms/version, including pandas (which is 11.7 MB, xarray is 870 kB).

(There is another xarray-test above, so two.)

I can of course remove them, but I think we should discuss if we want to support more formats than pandas and numpy (and Python list/tuple), and, if so, have explicit tests for them.

Reply:
I agree, we should probably discuss what we want to support/test. To me, this doesn't seem to add a whole lot of value for adding a new dependency.

There was also a discussion around removing Scipy as a  dependency in the docs: https://github.com/matplotlib/matplotlib/pull/22120